force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -110,25 +124,34 @@ public with sharing class FlowControlLoopsCollections {
      * @return The grade for the given score.
      */
     public static String determineGrade(Integer score) {
-        return null; // Replace null with the variable you used to store the result
+        if (score >= 90) {
+            return 'A';
+        } else if (score >= 80) {
+            return 'B';
+        } else if (score >= 70) {
+            return 'C';
+        } else if (score >= 60) {
+            return 'D';
+        } else {
+            return 'F';
+        }
     }
 
 
     /**
      * Question 8
      * Sum all the integers up to a given limit.
      * Example: sumUpToLimit(5) should return 15
-     * @param intLimit The number up to which integers will be summed.
+     * @param intLimit The number up to which integers will be summed.  
      * @return The sum of all integers up to the limit.
      */
     public static Integer sumUpToLimit(Integer intLimit) {
-        /* 
-         for (){
-            
-         }
-        */
+        // NOTE: Normally I wouldn't use recurssion to solve a problem like this, but Brooks suggested to try and use recurssion for this specific question
+        if (intLimit <= 0) {
+            return 0;  
+        }
 
-        return null; // Replace null with the variable you used to store the result
+        return intLimit + sumUpToLimit(intLimit - 1);
```

Comment:
Yep; there are programming languages where you DO want to code like this. 
Take a look at functional or procedural programming languages like OCaml if youre truly curious. 
In general, this is not "idiomatic Apex". A lot of people for some reason struggle with recursion, and so we use loops instead. I think they are more "natural" to read but I personally don't mind either approach.